### ISSUE TITLE
Don't report __orig_class__.__args__ as unknown members for generics

### ISSUE THREAD

**Issue by neijrr** [DetachHead/basedpyright#1573]
### Description

While it is an implementation detail, currently it's the only way to reliably get access to type (instead of TypeVar) for generic class/function.
So:
- In non-generic classes/functions (so `class X(...)` or `def x(...)`) `__orig_class__` should be reported as unknown, since it does not exist there
- In generic classes/functions (`class Y[...](...)` or `def y[...](...)`) `__orig_class__` should _not_ be reported as unknown, and `__orig_class__.__args__` should be considered to be `tuple[type, ...]`

Example:
[basedpyright playground](https://basedpyright.com/?typeCheckingMode=all&code=MYGwhgzhAECCDaAVAugLgFDS9AJgUwDNoCBXAO2AAoI8QCAaaMVaASzIBdoAfYkAezBdeiAJTQAtAD42nDNgXQAxNADCACzzAA1myJg2MdhA5gKeaPyKJMirKyKsIx0%2BcphGNOgDoA%2Br-4AJ1YAc19QSAh-P18wQJCo33gABmRReTtFQLwOEkCyaGTbOwdDFzNgPHdGdg504sysbNz86ABGBoVmvIKAJnQB5jh4E0DkaABeYdHkSlF0AAdgzndvUgpKACIwTdF5pdrV9ao2vcXljiPyE%2B9T%2BaA)
```python
class A[T]:
    def func(self, a: int | float | T) -> int:
        # Check if a is instance of T
        if isinstance(a, self.__orig_class__.__args__[0]):
            return 0
        if isinstance(a, int):
            return 1
        return 2


a: A[str] = A[str]()
print(a.func("a"))
print(a.func(1))
print(a.func(1.1))
```

**Comment by DetachHead** [comment]
the issue with not reporting an error here is we can't guarantee that `__orig_class__` will actually be there at runtime:

```py
a: A[str] = A()
print(a.func("a"))
```
```
AttributeError: 'A' object has no attribute '__orig_class__'. Did you mean: '__orig_bases__'?
```
